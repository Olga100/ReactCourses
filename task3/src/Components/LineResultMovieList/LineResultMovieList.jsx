import React from 'react';
import './LineResultMovieList.scss';

const LineResultMovieList = ({countMovies, sortBy}) => {
    if( countMovies  != null &&  sortBy  != null) {
        console.log(countMovies);
        return (
        <div className="lineResultContainer">
            <span>{countMovies} movies found</span>
            <span> Sort by  {sortBy}</span>
        </div>
        )
    } else {
        console.log("lllll")
        return (<div className="lineResultContainer"></div>)
    }
};




export default LineResultMovieList;