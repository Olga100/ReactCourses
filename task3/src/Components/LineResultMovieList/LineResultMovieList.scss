.lineResultContainer {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 20px;
  font-size: 14px;
  font-weight: bold;
  background-color: #f5f5f5;

  span:first-child {
    margin-left: 35px;
  }
  .sortByWrapper {
    span {
      margin-right: 35px;
      cursor: pointer;
    }
    span:first-child {
      margin-right: 20px;
      cursor: default;
    }
    .active {
      color: #c15d69;
    }
  }
}
