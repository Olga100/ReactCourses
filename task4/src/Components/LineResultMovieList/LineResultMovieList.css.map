{
"version": 3,
"mappings": "AAAA,oBAAqB;EACnB,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,aAAa;EAC9B,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;EACZ,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,gBAAgB,EAAE,OAAO;EAEzB,qCAAiB;IACf,WAAW,EAAE,IAAI;EAGjB,wCAAK;IACH,YAAY,EAAE,IAAI;IAClB,MAAM,EAAE,OAAO;EAEjB,oDAAiB;IACf,YAAY,EAAE,IAAI;IAClB,MAAM,EAAE,OAAO;EAEjB,2CAAQ;IACN,KAAK,EAAE,OAAO",
"sources": ["LineResultMovieList.scss"],
"names": [],
"file": "LineResultMovieList.css"
}