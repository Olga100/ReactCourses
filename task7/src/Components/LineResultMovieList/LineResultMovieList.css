.lineResultContainer {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 20px;
  font-size: 14px;
  font-weight: bold;
  background-color: #f5f5f5; }
  .lineResultContainer span:first-child {
    margin-left: 35px; }
  .lineResultContainer .sortByWrapper span {
    margin-right: 35px;
    cursor: pointer; }
  .lineResultContainer .sortByWrapper span:first-child {
    margin-right: 20px;
    cursor: default; }
  .lineResultContainer .sortByWrapper .active {
    color: #c15d69; }

/*# sourceMappingURL=LineResultMovieList.css.map */
