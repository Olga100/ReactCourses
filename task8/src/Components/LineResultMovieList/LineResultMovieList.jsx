import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

//import './LineResultMovieList.scss';


const styles = {
    lineResultContainer: {
        display: 'flex',
        justifycontent: 'space-between',
        width: '100%',
        height: '20px',
        fontSize: '14px',
        fontWeight: 'bold',
        backgroundColor: '#f5f5f5'
    },
    sortByWrapper:{
        display: 'flex',
        width: '250px',
        justifyContent: 'space-around'
    },
    sortByWrapperSpan: {
            margin: '0 20px',
            cursor: 'pointer'
        },

    active: {
            color: '#c15d69'
        }
    }


const LineResultMovieList = ({moviesCount, sortBy, sortByChanged, classes}) => {
    const sortByReleaseDate = () => sortByChanged('release_date');
    const sortByRating = () => sortByChanged('vote_average');

    return moviesCount === null || sortBy === null
        ? null
        : <div className={classes.lineResultContainer}>
            <span>{moviesCount} movies found</span>
            <span className={classes.sortByWrapper}>
            <span className={classes.sortByWrapperSpan}>Sort by</span>
            <span
                id="sortByRelease"
                className={(sortBy === "release_date")?  classes.active  : null}
                onClick={sortByReleaseDate}
            >release date</span>
            <span
                  id="sortByRating"
                  className={(sortBy === "vote_average")?  classes.active  : null}
                  onClick={sortByRating}
            >rating</span>
            </span>
        </div>
};

LineResultMovieList.propTypes = {
    moviesCount: PropTypes.number,
    sortBy:  PropTypes.string,
    sortByChanged: PropTypes.func,
};

export default injectSheet(styles)(LineResultMovieList);
