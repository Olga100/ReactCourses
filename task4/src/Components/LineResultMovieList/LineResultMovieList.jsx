import React from 'react';
import PropTypes from 'prop-types';

import './LineResultMovieList.scss';

const LineResultMovieList = ({moviesCount, sortBy, sortByChanged}) => {
    const handleSortByDate = () => sortByChanged('release_date');
    const handleSortByVote = () => sortByChanged('vote_average');

    return moviesCount === null || sortBy === null
        ? null
        : <div className="lineResultContainer">
            <span>{moviesCount} movies found</span>
            <span className="sortByWrapper">
            <span>Sort by</span>
            <span
                id="sortByRelease"
                className={(sortBy === "release_date")? "active" : null}
                onClick={handleSortByDate}
            >release date</span>
            <span
                id="sortByRating"
                  className={(sortBy === "vote_average")? "active" : null}
                  onClick={handleSortByVote}
            >rating</span>
            </span>
        </div>
};

LineResultMovieList.propTypes = {
    moviesCount: PropTypes.number.isRequired,
    sortBy:  PropTypes.string.isRequired,
    sortByChanged: PropTypes.func,
};

export default LineResultMovieList;
