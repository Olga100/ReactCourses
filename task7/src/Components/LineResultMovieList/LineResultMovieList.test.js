import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import LineResultMovieList from './LineResultMovieList';


describe ('LineResultMovieList', () => {

    it('should match snapshot with data', () => {
        const component = renderer.create(
            <LineResultMovieList  moviesCount={3} sortBy={'release_date'} />
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it('should match snapshot without data', () => {
        const component = renderer.create(
            <LineResultMovieList  moviesCount={null} sortBy={null} />
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

    it('Calls callback "release date" on click', () => {
        const handleSortByDateSpy = jest.fn();
        const content = shallow(<LineResultMovieList  moviesCount={10} sortBy={'vote_average'} sortByChanged={handleSortByDateSpy} />);

        content.find('#sortByRelease').simulate('click');

        expect(handleSortByDateSpy).toHaveBeenCalledWith('release_date');
    });

    it('Calls callback "rating" on click', () => {
        const handleSortByVoteSpy = jest.fn();
        const content = shallow(<LineResultMovieList  moviesCount={10} sortBy={'release_date'} sortByChanged={handleSortByVoteSpy} />);

        content.find('#sortByRating').simulate('click');

        expect(handleSortByVoteSpy).toHaveBeenCalledWith('vote_average');
    });

});
