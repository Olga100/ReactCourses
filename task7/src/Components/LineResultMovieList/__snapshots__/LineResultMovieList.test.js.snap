// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineResultMovieList should match snapshot with data 1`] = `
<div
  className="lineResultContainer"
>
  <span>
    3
     movies found
  </span>
  <span
    className="sortByWrapper"
  >
    <span>
      Sort by
    </span>
    <span
      className="active"
      id="sortByRelease"
      onClick={[Function]}
    >
      release date
    </span>
    <span
      className={null}
      id="sortByRating"
      onClick={[Function]}
    >
      rating
    </span>
  </span>
</div>
`;

exports[`LineResultMovieList should match snapshot without data 1`] = `null`;
